Dockerfile.ci-rp

```diff
@@ -0,0 +1,38 @@
+ARG REGISTRY
+FROM ${REGISTRY}/ubi8/go-toolset:1.20.10 as builder
+
+USER root
+ENV GOPATH=/root/go
+RUN mkdir -p /app
+WORKDIR /app
+COPY . /app
+RUN make aro RELEASE=${IS_OFFICIAL_RELEASE} -o generate
+
+FROM builder as vendor-check
+WORKDIR /app
+COPY --from=builder /app/aro /app/e2e.test /usr/local/bin/
+RUN go mod vendor && go mod tidy -compat=1.18 && hack/ci-utils/isClean.sh
```

Comment:
Any reason to check [compatibility](https://go.dev/ref/mod#go-mod-tidy) for v1.18 and not the latest Golang version in the repo? Golang [v1.20](https://github.com/Azure/ARO-RP/blob/master/go.mod#L3) was also used in [this file](https://github.com/Azure/ARO-RP/pull/3489/files#diff-d189eef96a9753a0ef4403084a625155b556091a2b336d98e20f24139b7ac3acR2).

Reply:
Good callout; the bump to 1.20 was a very recent change. @tony-schndr can you rebase and get this working for 1.20?

Reply:
bumped